haystack/utils/import_utils.py

```diff
@@ -62,6 +64,28 @@ def _optional_component_not_installed(component: str, dep_group: str, source_err
     ) from source_error
 
 
+def load_documents_from_remote(dataset_name: str, split: Optional[str] = "train") -> List[Document]:
+    """
+    Load a list of Haystack Documents from a remote Hugging Face dataset.
+
+    :param dataset_name: A Hugging Face dataset containing Haystack Documents
+    :param split: Optional parameter. The split of the Hugging Face dataset to load from. By default this is set to "train".
+    :return a List of Documents
+    """
+    try:
+        dataset = load_dataset_builder(dataset_name)
+        document_keys = set(["content", "content_type"])
+        if not document_keys.issubset(dataset.info.features.keys()):
+            raise Exception("This dataset does not conntain Haystack Documents.")
```

Comment:
What all can go wrong? - 
1. If HF is down, would load_dataset_builder fail? If so let’s raise appropriate error saying HF is down. I am not able to find an example but if you don’t find it either, Let me know I will try to create one.
2. Let’s raise HaystackError, instead of generic Exception. Here is [an Example](https://github.com/deepset-ai/haystack/blob/3782ebc835a60b0126fb3ad5269e98e40d38d475/haystack/nodes/other/docs2answers.py#L43)